src/desktop/apps/artwork/client/errorModal.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react"
+import ReactDOM from "react-dom"
+import { ErrorModal } from "reaction/Components/Modal/ErrorModal"
+
+export const render = ({
+  headerText = "An error occurred",
+  detailText = "Something went wrong. Please try again or contact orders@artsy.net.",
+} = {}) => {
+  const el = document.getElementById("react-modal-container")
+  const onClose = () => {
+    ReactDOM.unmountComponentAtNode(el)
+  }
+  ReactDOM.render(
+    <ErrorModal
+      show={true}
+      onClose={onClose}
+      headerText={headerText}
+      detailText={detailText}
+    />,
+    el
+  )
+}
+
+interface ApplicationError {
+  code?: string
+}
+export const renderBuyNowError = (error: ApplicationError) => {
+  if (error && error.code) {
+    switch (error.code) {
```

Comment:
In this case, I don't think it's mandatory or a huge difference, but we can consider specifying the error codes in the interface that way

```typescript
interface ApplicationError {
  code?: "unpublished_artwork" | "unknown_artwork"
}
```

Reply:
Totally agree. I think the best solution here would be to update the graphql schema to use enums so that the types flow all the way down.